klarna-on-demand/src/main/java/com/klarna/ondemand/OriginProof.java

```diff
@@ -0,0 +1,64 @@
+package com.klarna.ondemand;
+
+import android.util.Base64;
+import org.json.JSONObject;
+import java.util.UUID;
+
+/**
+ * Assists in creating origin proofs that allow performing secure purchases.
```

Comment:
I think I'd go with singular, as in: "An origin proof that allows performing secure purchases."